src/BccCode.Linq/QueryProvider/ApiQueryProvider.cs

```diff
@@ -932,46 +1006,54 @@ protected override Expression VisitMethodCall(MethodCallExpression node)
                     return base.VisitMethodCall(node);
             }
         }
-        else if (node.Method.DeclaringType?.FullName == "Microsoft.EntityFrameworkCore.Query.IncludeExpression" &&
-                 node.Method.Name == "Include")
+        else if (node.Method.DeclaringType == typeof(QueryableExtensions))
         {
-            //Arg 0: source
-            var source = Visit(node.Arguments[0]);
-            Debug.Assert(source != null);
-            if (!TryGetApiCaller(node.Arguments[0], out var apiCaller))
-            {
-                // ... the source was not a ApiQueryable and is not
-                // part of the API. So we just do nothing and pass it
-            }
-            else if (node.Arguments.Count == 2)
+            switch (node.Method.Name)
             {
-                if (node.Arguments[1] is UnaryExpression u && u.Operand is LambdaExpression l)
+                case nameof(QueryableExtensions.Include):
                 {
-                    var sourceType = node.Arguments[0].Type.GenericTypeArguments[0];
-
-                    var declaringType = l.Parameters[0].Type;
-                    if (declaringType != sourceType)
+                    //Arg 0: source
+                    var source = Visit(node.Arguments[0]);
+                    Debug.Assert(source != null);
+                    if (!TryGetApiCaller(node.Arguments[0], out var apiCaller))
                     {
-                        throw new Exception($"Expanding is only supported on root document type {sourceType.Name}");
+                        // ... the source was not a ApiQueryable and is not
+                        // part of the API. So we just do nothing and pass it
                     }
+                    else if (node.Arguments.Count == 2)
+                    {
+                        if (node.Arguments[1] is UnaryExpression u && u.Operand is LambdaExpression l)
+                        {
+                            var sourceType = node.Arguments[0].Type.GenericTypeArguments[0];
+
+                            var declaringType = l.Parameters[0].Type;
+                            if (declaringType != sourceType)
+                            {
```

Comment:
Would it be difficult to expand on a nested object. E.g. with a ThenInclude(...) method?
This is at least something that the API supports, and has been requested by developers.

Reply:
I had already in mind to implement that, but decided to skip it for now. I created issue #40 for it.